pkgs/applications/office/paperless/default.nix

```
@@ -0,0 +1,222 @@
+{ stdenv
+, lib
+, writeScript
+, fetchFromGitHub
+
+, python36
+, file
+, imagemagickBig # includes ghostscript
+, optipng
+, poppler
+, tesseract
+, unpaper
+
+, enableTests ? true
+}:
+
+## Usage
+#
+# ${pkgs.paperless}/bin/paperless wraps manage.py
+#
+# ${pkgs.paperless}/share/paperless/setup-env.sh can be sourced from a
+#   shell script to setup a Paperless environment
+#
+# paperless.withConfig is a convenience function to setup a
+# configured Paperless instance. Example:
+#
+# nix-build --out-link ./paperless -E '
+# (import <nixpkgs> {}).paperless.withConfig {
+#   dataDir = /tmp/paperless-data;
+#   config = {
+#     PAPERLESS_DISABLE_LOGIN = "true";
+#   };
+# }'
+#
+## Setup DB
+# ./paperless migrate
+#
+## Consume documents in dataDir/consume
+# ./paperless document_consumer --oneshot
+#
+## Start web interface
+# ./paperless runserver --noreload localhost:8000
+
+
+## WSGI with gunicorn
+# Use a shell script like this:
+# source ${paperless}/share/paperless/setup-env.sh
+# PYTHONPATH=$paperlessSrc gunicorn paperless.wsgi
+
+with lib;
+let
+  paperless = stdenv.mkDerivation rec {
+    name = "paperless-${version}";
+    # Don't use 2.5.0, the latest official release, because it's broken on macOS
+    # (Paperless issue #418)
+    version = "2.6.0-pre";
+
+    src = fetchFromGitHub {
+      owner = "danielquinn";
+      repo = "paperless";
+      rev = "c83dc666a448ee7be8faa970313be0f68bf53a19";
+      sha256 = "1p8519hqs6k0b7pglpkabkl0cmgzj84dq4sabnhffi0r44z2mazw";
+    };
+
+    inherit python;
+
+    pythonEnv = python.withPackages (ps: with ps; [
+      coveralls
+      dateparser
+      dateutil
+      django
+      django-crispy-forms
+      django-cors-headers
+      django-filter
+      django_extensions
+      djangorestframework
+      factory_boy
+      filemagic
+      flake8
+      fuzzywuzzy
+      gunicorn
+      langdetect
+      pdftotext
+      pillow
+      pycodestyle
+      pyocrWithUserTesseract
+      pytest
+      pytest-django
+      pytest-env
+      # pytest_xdist # Disabled, broken in nixpkgs
+      pytestcov
+      python-dotenv
+      python-gnupg
+      pytz
+      termcolor
+    ] ++ (optional stdenv.isLinux inotify_simple));
+
+    buildCommand = let
+      # Paperless has explicit runtime checks that expect these binaries to be in PATH
+      extraBin = makeBinPath [ imagemagickBig optipng tesseract unpaper ];
+    in ''
+      srcDir=$out/share/paperless
+      mkdir -p $out/bin $out/share
+      cp -r $src/src $srcDir
+      chmod -R +w $srcDir
+      cp $src/LICENSE $srcDir
+
+      ### Patches
+
+      # Remove command arguments for pytest_xdist, which is broken in nixpkgs
+      chmod +w $srcDir/setup.cfg
+      sed -i 's/ -n auto//' $srcDir/setup.cfg
+
+      ### Compile
+
+      ${python}/bin/python -m compileall $srcDir
```

```suggestion
      ${python.interpreter} -m compileall $srcDir
```
The same multiple times.

---

Done.